core/src/commonMain/kotlin/Peripheral.kt

```diff
@@ -33,38 +35,44 @@ public interface Peripheral {
      *                      | Disconnecting | ----> | Disconnected |
      *                      '---------------'       '--------------'
      * ```
-     *
-     * The [state] [Flow] is conflated (so some states may be missed by slow consumers); it is intended to provide the
-     * current connection status to the user. If it is desired to handle (take action based on) specific connection
-     * events, then [events] [Flow] should be used instead.
      */
     public val state: Flow<State>
 
-    public val events: Flow<Event>
-
     /**
      * Initiates a connection, suspending until connected, or failure occurs. Multiple concurrent invocations will all
      * suspend until connected (or failure occurs). If already connected, then returns immediately.
      *
+     * @throws ConnectionRejectedException when a connection request is rejected by the system (e.g. bluetooth hardware unavailable).
      * @throws IllegalStateException if [Peripheral]'s Coroutine scope has been cancelled.
      */
     public suspend fun connect(): Unit
 
+    /**
+     * Disconnects the active connection, or cancels an in-flight [connection][connect] attempt, suspending until
+     * [Peripheral] has settled on a [disconnected][State.Disconnected] state.
+     *
+     * Multiple concurrent invocations will all suspend until disconnected (or failure occurs).
+     */
+    public suspend fun disconnect(): Unit
+
     /** @return discovered [services][Service], or `null` until a [connection][connect] has been established. */
     public val services: List<DiscoveredService>?
 
+    @Throws(CancellationException::class, IOException::class)
     public suspend fun rssi(): Int
 
     /**
      * @throws NotReadyException if invoked without an established [connection][connect].
      */
+    @Throws(CancellationException::class, IOException::class)
     public suspend fun read(
         characteristic: Characteristic,
     ): ByteArray
 
     /**
      * @throws NotReadyException if invoked without an established [connection][connect].
      */
+    @Throws(CancellationException::class, IOException::class)
```

Comment:
Was all an oversight. Needed the `@Throws` annotations as they are used for proper interop with Apple platforms (on the consumer side).

I'll go over these and make sure they're all correct.

`@Throws` is the correct approach AFAIK, so I'll drop the KDoc `@throws` comments.

Reply:
Updated to at least have all the needed exceptions in the `@Throws`.

I went to JetBrains repos for guidance on how I should approach this, unfortunately none of the following projects use `@Throws`:
- kotlinx.coroutines
- kotlinx.serialization
- kotlinx.datetime

It looks like Square [uses the annotation in Okio](https://github.com/square/okio/blob/000e2dcbc6d855300afdf49b3e23faa42d56dd57/okio-files/src/commonMain/kotlin/okio/Filesystem.kt#L72-L73), so, I figure that's a good reference.

It looks like they use both `@Throws` and `@throws` to add further clarification in the KDoc.

I figure I'll follow that approach and add the exceptions to `@Throws` so that it's properly exposed on the public API for Apple-based platform consumers, then also have `@throws` KDoc for Exceptions that should have further explanation. For example, `NotReadyException` is specific to Kable so warrants further explanation whereas `IOException` is pretty much universally understood (so no need to add KDoc for it, IMHO).